aten/src/ATen/TensorIterator.h

```diff
@@ -476,6 +476,7 @@ class TORCH_API TensorIteratorConfig final {
   //   the operation.
   // Setting this flag to true sets check_all_same_dtype_ to false.
   TensorIteratorConfig& promote_inputs_to_common_dtype(const bool _promote_inputs_to_common_dtype);
+  TensorIteratorConfig& set_common_dtype(c10::optional<ScalarType> common_dtype);
```

Comment:
Add a comment describing what this setting does.

Why is the input an optional?

Reply:
This comment will need to be updated (slightly) to reflect the can_cast check.